ddl/placement/rule_test.go

```diff
@@ -50,15 +48,32 @@ func matchRules(t1, t2 []*Rule, prefix string, c *C) {
 	}
 }
 
-func (t *testRuleSuite) TestNewRuleAndNewRules(c *C) {
+func matchRulesT(t1, t2 []*Rule, prefix string, t *testing.T) {
+	require.Equal(t, len(t2), len(t1), prefix)
+	for i := range t1 {
+		found := false
+		for j := range t2 {
+			ok, _ := DeepEquals.Check([]interface{}{t2[j], t1[i]}, []string{})
```

Comment:
DeepEquals depend on `check`, could we remove it here too

Reply:
Done